kickstarter/android-oss
# PAY-1870:  Update pledge flows with PaymentSheet

# 📲 What

- Stripe's PaymentSheet added to the pledge and update pledge flows
- ℹ️ Additionally removed the codecoverage integration as currently failing showing 0%, and the internal builds keep failing at this point @hadia or @leighdouglas feel free to create a ticket or investigate while I'm on PTO


# 👀 See
- 103 tests Ok for `PledgeFragmentViewModel`
<img width="916" alt="Screen Shot 2022-10-05 at 11 04 55 AM" src="https://user-images.githubusercontent.com/4083656/194023658-bc4b7374-cb81-4d90-9170-4fe49472d062.png">

- Pledge & change payment method flows 

https://user-images.githubusercontent.com/4083656/194020961-b1877da1-7a73-4a8d-a8bf-1c1d9f09ba8a.mp4


|  |  |

# 📋 QA

⚠️  remember that for being able to use the update flows the project needs to be still going (Screenshot attached), meaning a project with "0 seconds to go" is no valid. 
<img width="427" alt="Screen Shot 2022-10-05 at 10 54 46 AM" src="https://user-images.githubusercontent.com/4083656/194021806-10a9a47a-52d4-4a9f-9693-c71710a4b7e5.png">

- Pledge to a project
- Update a pledge
- Change a reward
- Change payment method -> add a new payment method using the PaymentSheet
- Try to Fix a pledge ⚠️ tested on production with my personal account, but having the same situation as described below 👇 

@lukert33 or @iambchan in the video you can see as well that after successfully changing the payment method (calling UpdateBacking mutation) the backing information seems to not be updated with the new payment method, or it added to the user payment methods either.

# Story 📖

[PAY-1870](https://kickstarter.atlassian.net/browse/PAY-1870)
